e2e/backend.e2e.spec.ts

```diff
@@ -0,0 +1,228 @@
+        '--disable-gpu',
+        '--whitelisted-ips=',
+        // See src/utils/commandLine.ts before changing the next item as the final option.
+        '--disable-dev-shm-usage',
+        '--no-modal-dialogs',
+      ],
+      env: {
+        ...process.env,
+        RD_LOGS_DIR: reportAsset(__filename, 'log'),
+      },
+    });
+    context = electronApp.context();
+
+    await context.tracing.start({ screenshots: true, snapshots: true });
+    page = await electronApp.firstWindow();
+  });
+
+  test.afterAll(async() => {
+    await context.tracing.stop({ path: reportAsset(__filename) });
+    await packageLogs(__filename);
+    await electronApp.close();
+  });
+
+  test('should start loading the background services and hide progress bar', async() => {
+    const navPage = new NavPage(page);
+
+    await navPage.progressBecomesReady();
+    await expect(navPage.progressBar).toBeHidden();
+  });
+
+  test.describe('requiresRestartReasons', () => {
+    let serverState: { user: string, password: string, port: string, pid: string };
+
+    test.afterEach(async() => {
+      // Wait for the backend to stop (it's okay to fail to start here though)
+      const navPage = new NavPage(page);
+
+      while (await navPage.progressBar.count() > 0) {
+        await navPage.progressBar.waitFor({ state: 'detached', timeout: 120_000 });
+      }
+    });
+
+    test('should emit connection information', async() => {
+      const dataPath = path.join(paths.appHome, 'rd-engine.json');
+      const dataRaw = await fs.promises.readFile(dataPath, 'utf-8');
+
+      serverState = JSON.parse(dataRaw);
+      expect(serverState).toEqual(expect.objectContaining({
+        user:     expect.any(String),
+        password: expect.any(String),
+        port:     expect.any(Number),
+        pid:      expect.any(Number),
+      }));
+    });
+
+    async function get(requestPath: string) {
+      const auth = Buffer.from(`${ serverState.user }:${ serverState.password }`).toString('base64');
+      const result = await fetch(`http://127.0.0.1:${ serverState.port }/${ requestPath.replace(/^\//, '') }`, { headers: { Authorization: `basic ${ auth }` } });
+
+      expect(result).toEqual(expect.objectContaining({ ok: true }));
+
+      return await result.json();
+    }
+
+    /**
+     * getOldSettings returns the subset of settings that are being modified.
+     */
+    function getOldSettings(oldSettings: Settings, newSettings: RecursivePartial<Settings>): RecursivePartial<Settings> {
+      /**
+       * keysDeep returns the list of paths of all elements of the given object.
+       */
+      const keysDeep: (obj: any, path: string[]) => string[][] = (obj, path = []) => {
+        return Object.keys(obj).flatMap((k) => {
+          return _.isObjectLike(obj[k]) ? keysDeep(obj[k], path.concat(k)) : [path.concat(k)];
+        });
+      };
+
+      // The typing for _.pick is incorrect: paths are a (string | string[])[] argument, not a rest argument.
```

Comment:
I don't understand this comment. Is it suggesting an action we need to carry out, or a problem with the upstream `lodash.pick` typescript def'n? Also neither this function nor the one I landed on by doing `show-definition-for _.pick` has something called called `paths`.

Reply:
The TypeScript definition (in [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/7b22711d30a366a28a698353e2cf68cc102a4f09/types/lodash/common/object.d.ts#L2004)) does not match the actual observed behaviour.

Though I think the comment might be wrong anyways; looking into it more.